src/nvim/ex_getln.c

```diff
@@ -2161,6 +2175,14 @@ static void draw_cmdline(int start, int len)
 {
   int i;
 
+  if (cmdline_external) {
+    Array args = ARRAY_DICT_INIT;
+    ADD(args, STRING_OBJ(cstr_to_string((char *)(ccline.cmdbuff))));
+    ADD(args, INTEGER_OBJ(ccline.cmdpos));
```

Comment:
@bfredl It looks like interaction with #6364 is going to be pretty straightforward: move `color_cmdline` call to the start of the function, stuff the results here. Only need to agree now how these results are transferred.

The only thing which bothers me is coloring of arabic shaping: I have no tests for this (if I am not mistaking, new Vim tests were not ported) and not sure what I need to do in first place.